### PR TITLE
Get property ID for SSA step assert

### PR DESCRIPTION
**Pull request by peterschrammel**
Make reusable to avoid error-prone duplication.

- [X] Each commit message has a non-empty body, explaining why the change was made.
- [X] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: f8f6cdbac03ae4a287356e0c9779bc6b5b1463f4).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/97472573

**Review by smowton — APPROVED** [review]

File: src/cbmc/all_properties.cpp [hunk 1]
```diff
@@ -78,20 +78,16 @@ safety_checkert::resultt bmc_all_propertiest::operator()()
...
 
-      if(it->source.pc->is_assert())
-        property_id=it->source.pc->source_location.get_property_id();
-      else if(it->source.pc->is_goto())
+      if(property_id.empty())
+        continue;
+
+      if(it->source.pc->is_goto())
       {
         // this is likely an unwinding assertion
```
Comment (tautschnig) [inline src/cbmc/all_properties.cpp]: I think this comment no longer is a good fit here. Maybe put it before the `if` and say "goto may yield and unwinding assertion"?

**Review by tautschnig — APPROVED** [review]
Could the rationale from the PR description please also be copied into the commit message?

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 9c48c53fb89dd6e7ccca5ad061a3af808d29272d).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/97530900

**Pull request merged**